Formula/fheroes2.rb

```diff
@@ -0,0 +1,38 @@
+class Fheroes2 < Formula
+  desc "Free Heroes of Might and Magic II () is a recreation of HoMM2 game engine"
+  homepage "https://ihhub.github.io/fheroes2/"
+  url "https://github.com/ihhub/fheroes2/archive/refs/tags/0.9.7.tar.gz"
+  sha256 "02e777485906b8112636173c7e241d2824b2dce7707b5ebf6b3bab70ea9865f7"
+  license "GPL-2.0-or-later"
+  head "https://github.com/ihhub/fheroes2.git", branch: "master"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "cmake" => :build
+  depends_on "gettext" => :build
+
+  depends_on "libpng"
+  depends_on "sdl2"
+  depends_on "sdl2_image"
+  depends_on "sdl2_mixer"
+
+  uses_from_macos "zlib"
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+    system "make", "-C", "files/lang"
+
+    (pkgshare/"files/lang").install Dir["files/lang/*.mo"]
+    pkgshare.install "fheroes2.key"
+    bin.install "script/demo/demo_unix.sh" => "fheroes2-install-demo"
+  end
+
+  test do
+    assert_match "help", shell_output("#{bin}/fheroes2 -h")
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
There isn't many options:

```
Usage: fheroes2 [OPTIONS]
  -h		print this help message and exit
```

Reply:
The real test will require GUI, since it's a game.

Reply:
Shouldn't it be added as a cask then? It seems they provide a binary for macOS: https://github.com/ihhub/fheroes2/releases/download/0.9.7/fheroes2_macos10_15_sdl2.zip

Reply:
Yes, you are right, I didn't read [Acceptable Casks](https://docs.brew.sh/Acceptable-Casks) doc carefully.

>The app is both open-source and _CLI-only_ (i.e. it only uses the binary artifact).

It seems like cask is a better form / home for this app.

Reply:
There is no GUI .app there, it has the same CLI launcher binary, that is being built here via formula.

Reply:
Can we do something like the test in `allureofthestars`?

Reply:
Unfortunately I don't know proper way to test it. It's only engine, resources can be extracted from demo version of original game by script `fheroes2-install-demo`, but it uses `~/.fheroes2` path, which we can't use to due to sandbox limitations.